components/select_team/components/select_team_item.jsx

```diff
@@ -59,14 +58,14 @@ export default class SelectTeamItem extends React.PureComponent {
         return (
             <div className='signup-team-dir'>
                 {showDescriptionTooltip}
-                <Link
-                    to={`/${this.props.team.name}/channels/town-square`}
+                <a
```

Comment:
You don't need to switch to an a component to make this work. React-router Link component supports onClick. I doubt this fix.

Reply:
Link had `onClick`, but at the same time `to`, and the `to` generate a bypass over the `onClick`, but I'm now realizing probably what I need is a preventDefault in the onClick event. I going to try it.

Reply:
But are the `onclick` and the `<Link/>` both handling the router? That does seem like it could add unneeded complexity, and more chance for breakage?
Although that was how it was before, so I am approving because this fixes the bug.

Reply:
Actually with the `preventDefault` is only the `onClick` managing the route change. Before that, the `to` and the `onClick` were making a race to see who wins. I don't know which is the best approach here, but the `to` field is required in `<Link/>` component, and is not bad to have it as a "documentation" of what is the expected behavior. But I'm 0/5 about it.

Reply:
Except that it doesn't... onClick is the first thing executed by link: https://github.com/ReactTraining/react-router/blob/master/packages/react-router-dom/modules/Link.js#L36

If you want to stop the transition, then you can prevent default. Not sure that's what your going for.

Reply:
Yes, exactly, is what I'm doing now, in my last push.